src/VirtoCommerce.NotificationsModule.Web/Scripts/blades/notification-templates-list.js

```diff
@@ -1,155 +1,187 @@
-                        if (remove) {
-                            bladeNavigationService.closeChildrenBlades(blade, function () {
-                                _.each(selections,function(selection) {
-                                    var index = blade.currentEntity.templates.findIndex(function (element) {
-                                        return (element.languageCode === selection.languageCode);
-                                    });
-        
-                                    if (index > -1) {
-                                        blade.currentEntity.templates.splice(index, 1);
-                                    }
-                                });
-                            });
+
+            function deleteList(selections) {
+                var containsDefaultNotifications = _.any(selections, function (item) {
+                    return !item.languageCode || item.isPredefined;
+                });
+
+                if (containsDefaultNotifications) {
+                    dialogService.showNotificationDialog({
+                        id: "error",
+                        title: "notifications.dialogs.notification-template-delete-notification.title",
+                        message: "notifications.dialogs.notification-template-delete-notification.message"
+                    });
+                }
+                else {
+                    var dialog = {
+                        id: "confirmDeleteItem",
+                        title: "notifications.dialogs.notification-template-delete.title",
+                        message: "notifications.dialogs.notification-template-delete.message",
+                        callback: function (remove) {
+                            if (remove) { deleteFromGrid(); }
                         }
-                    }
-                };
-                dialogService.showConfirmationDialog(dialog);
+                    };
+                    dialogService.showConfirmationDialog(dialog);
+                }
             }
-            
-        }
-
-	    blade.toolbarCommands = [
-			    {
-				    name: "platform.commands.add", icon: 'fa fa-plus',
-				    executeMethod: createTemplate,
-				    canExecuteMethod: function () { return true; },
-				    permission: 'notifications:template:create'
-                },
-                {
-                    name: "platform.commands.delete", icon: 'fa fa-trash-o',
-                    executeMethod: function () { deleteList($scope.gridApi.selection.getSelectedRows()); },
-                    canExecuteMethod: function () {
-                        return $scope.gridApi && _.any($scope.gridApi.selection.getSelectedRows());
-                    },
-                    permission: 'notifications:template:delete'
+
+            function resetList(selections) {
+                var containsPostDefneidNotifications = _.any(selections, function (item) {
+                    return !item.isEdited || !item.isPredefined;
```

Comment:
probably it would be simpler with
```
item.isEdited && item.isPredefined
```

Reply:
Reversed version of `!item.isEdited || !item.isPredefined` is `!(item.isEdited && item.isPredefined)`. First variant is faster, because second forced us to check `item.isPredefined` for every item if `item.isEdited === true`. 

Btw, I don't like put round brackets 😄

Reply:
But `containsPostDefneidNotifications` means you have at least one predefined and edited at the same time.
So no negation needed based on the variable name. Maybe it is incorrect? Though I'd lease it as is (positive) and change usage in the `if` below
Btw typo in containsPostDe**fnei**dNotifications